src/components/user-info/user-info-controller.ts

```diff
@@ -32,6 +48,99 @@ export const userInfoController = async (
     phone_number_verified: config.getPhoneNumberVerified(),
   };
 
+  const claims = validationResult.claims;
+  tryAddClaim(
+    userInfo,
+    claims,
+    "https://vocab.account.gov.uk/v1/drivingPermit",
+    config.getDrivingPermitDetails()
+  );
+  tryAddClaim(
+    userInfo,
+    claims,
+    "https://vocab.account.gov.uk/v1/passport",
+    config.getPassportDetails()
+  );
+  tryAddClaim(
+    userInfo,
+    claims,
+    "https://vocab.account.gov.uk/v1/socialSecurityRecord",
+    config.getSocialSecurityRecordDetails()
+  );
+  tryAddClaim(
+    userInfo,
+    claims,
+    "https://vocab.account.gov.uk/v1/address",
+    config.getPostalAddressDetails()
+  );
+  tryAddClaim(
+    userInfo,
+    claims,
+    "https://vocab.account.gov.uk/v1/returnCode",
+    config.getReturnCodes()
+  );
+  await tryAddCoreIdentityJwt(userInfo, claims, config);
+
   res.status(200);
   res.send(userInfo);
 };
+
+const tryAddClaim = (
+  userInfo: UserIdentity,
+  requestedClaims: UserIdentityClaim[],
+  claim: UserIdentityClaim,
+  claimData: any
+): void => {
+  if (requestedClaims.includes(claim)) {
+    if (!claimData) {
+      logger.warn(
+        `Claim "${claim}" present in access token but not configured - ignored.`
+      );
+      return;
+    }
+
+    userInfo[claim] = claimData;
+  }
+};
+
+const tryAddCoreIdentityJwt = async (
+  userInfo: UserInfo,
+  requestedClaims: UserIdentityClaim[],
+  config: Config
+): Promise<void> => {
+  const claim: UserIdentityClaim =
+    "https://vocab.account.gov.uk/v1/coreIdentityJWT";
+  const vc = config.getVerifiableIdentityCredentials();
+
+  if (requestedClaims.includes(claim)) {
+    if (!vc) {
+      logger.warn(
+        `Claim "${claim}" present in access token but vc not configured - ignored.`
+      );
+      return;
+    }
```

Comment:
Ah yes this is what I was a bit concerned about re logging - I had an older version checked out, and it would error here if you didn't have a vc set, and the error had no logging 😅 so this looks good